### PR TITLE
Support workload identity federation and service principals

### LINKED ISSUE #46: Use Azure AD to authenticate to Azure App Configuration
Hello,

for more secured scenarios where Access Keys are disabled in Azure App Configuration instance there should be option to authenticate with Azure AD (for example Managed Identities) and not only with Connection String.

Greetings.

### PR DESCRIPTION
**Pull request by Botje**
This PR adds support for using App Config with a federated identity and service principals. (via client secret)
The necessary inputs are styled after Azure/login.

### CODE REVIEW

**Comment by Botje** [comment]
@microsoft-github-policy-service agree

**Comment by Botje** [comment]
Any chance this can get a review?

**Comment by ChristineWanjau** [comment]
@Botje Please pull the latest changes from master branch

File: src/input.ts [hunk 1]
```diff
@@ -27,15 +27,51 @@ export interface Input {
     contentType?: string;
 }
 
+export interface ConnectionString {
+    type: 'connection-string';
+    connectionString: string;
+}
+
+export interface Identity {
```
Comment (ChristineWanjau) [inline src/input.ts:35]: I would prefer if we rename this to WorkloadIdentity, instead of Identity. ServicePrincipal is also an Identity

Comment (ChristineWanjau) [reply src/input.ts:35]: Also change the type

**Comment by Botje** [comment]
I made the changes you suggested. Do you think it is worth creating a test for the "no `auth-type` set" scenario?

**Comment by ChristineWanjau** [comment]
@Botje I am just curious, are there any github actions that have used @azure/identity sdk to authenticate to azure? I see most of them recommend the use of @Azure/login.

**Comment by Botje** [comment]
I cannot find an NPM package named `@Azure/login`. Are you talking about the GitHub Action named `Azure/login`?  
That action uses the `az cli` to log in, and many actions defer authentication to that, yes.

If you prefer that can be added as a fourth authentication mechanism.

**Comment by ChristineWanjau** [comment]
Hello @Botje we are deprecating this Github action. We recommend using Azure CLI action instead. Follow this guideline to https://learn.microsoft.com/en-us/azure/azure-app-configuration/push-kv-github-action on how to set it up. I am going to go ahead and close this pull request.

**Pull request closed without merging**